captum/optim/_param/image/images.py

```diff
@@ -1,538 +1,484 @@
+    @staticmethod
+    def __new__(
+        cls: Type["ImageTensor"],
+        x: Union[List, np.ndarray, torch.Tensor] = [],
+        *args,
+        **kwargs,
+    ) -> torch.Tensor:
+        if isinstance(x, torch.Tensor) and x.is_cuda:
+            x.show = MethodType(cls.show, x)
+            x.export = MethodType(cls.export, x)
+            return x
+        else:
+            return super().__new__(cls, x, *args, **kwargs)
+
+    @classmethod
+    def open(cls, path: str, scale: float = 255.0) -> "ImageTensor":
+        if path.startswith("https://") or path.startswith("http://"):
+            response = requests.get(path, stream=True)
+            img = Image.open(response.raw)
+        else:
+            img = Image.open(path)
+        img_np = np.array(img.convert("RGB")).astype(np.float32)
+        return cls(img_np.transpose(2, 0, 1) / scale)
+
+    def __repr__(self) -> str:
+        prefix = "ImageTensor("
+        indent = len(prefix)
+        tensor_str = torch._tensor_str._tensor_str(self, indent)
+        suffixes = []
+        if self.device.type != torch._C._get_default_device() or (
+            self.device.type == "cuda"
+            and torch.cuda.current_device() != self.device.index
+        ):
+            suffixes.append("device='" + str(self.device) + "'")
+        return torch._tensor_str._add_suffixes(
+            prefix + tensor_str, suffixes, indent, force_newline=self.is_sparse
+        )
+
+    @classmethod
+    def __torch_function__(
+        cls: Type["ImageTensor"],
+        func: Callable,
+        types: List[Type[torch.Tensor]],
+        args: Tuple = (),
+        kwargs: dict = None,
+    ) -> torch.Tensor:
+        if kwargs is None:
+            kwargs = {}
+        return super().__torch_function__(func, types, args, kwargs)
+
+    def show(
+        self, figsize: Optional[Tuple[int, int]] = None, scale: float = 255.0
+    ) -> None:
+        show(self, figsize=figsize, scale=scale)
+
+    def export(self, filename: str, scale: float = 255.0) -> None:
+        save_tensor_as_image(self, filename=filename, scale=scale)
+
+
+class InputParameterization(torch.nn.Module):
+    def forward(self) -> torch.Tensor:
+        raise NotImplementedError
+
+
+class ImageParameterization(InputParameterization):
+    pass
+
+
+class FFTImage(ImageParameterization):
+    """Parameterize an image using inverse real 2D FFT"""
+
+    def __init__(
+        self,
+        size: Tuple[int, int] = None,
+        channels: int = 3,
+        batch: int = 1,
+        init: Optional[torch.Tensor] = None,
+    ) -> None:
+        super().__init__()
+        if init is None:
+            assert len(size) == 2
+            self.size = size
+        else:
+            assert init.dim() == 3 or init.dim() == 4
+            if init.dim() == 3:
+                init = init.unsqueeze(0)
+            self.size = (init.size(2), init.size(3))
+        self.torch_rfft, self.torch_irfft, self.torch_fft_freq = self.get_fft_funcs()
```

Comment:
nit: I think that `self.torch_fftfreq` might be a better variable name `self.torch_fft_freq` as it matches the function name.